### PR TITLE
F2: ProductList Page

### PR DESCRIPTION
**Pull request by sushmita12345**
Added ProductList page to the E-commerce website.

Screenshot 2022-03-21 at 12 55 23 PM

Screenshot 2022-03-21 at 12 55 47 PM

### CODE REVIEW

File: src/assets/Image/ProductListCard/index.js [hunk 1]
```diff
@@ -0,0 +1,70 @@
...
+// export {Image53} from "../../../assets/Image/ProductListCard/img53.webp";
+// export {Image54} from "../../../assets/Image/ProductListCard/img54.webp";
+// export {Image55} from "../../../assets/Image/ProductListCard/img55.webp";
+// export {Image56} from "../../../assets/Image/ProductListCard/img56.webp";
+// export {Image57} from "../../../assets/Image/ProductListCard/img57.webp";
+// export {Image58} from "../../../assets/Image/ProductListCard/img58.webp";
+// export {Image59} from "../../../assets/Image/ProductListCard/img59.webp";
+// export {Image60} from "../../../assets/Image/ProductListCard/img60.webp";
+
+// // export {Image1, Image2, Image3, Image4, Image5, Image6, Image7, Image8, Image9, Image10, Image, Image, Image, Image, Image, Image, Image, Image, Image, Image, Image, Image, Image, Image, Image, Image, Image, Image, Image, Image, Image, Image, Image, Image, Image, Image}
```
Comment (shraddha15-creator) [inline src/assets/Image/ProductListCard/index.js:62]: If commented code is not needed then you can delete it

File: src/assets/Image/ProductListCard/index.js [hunk 2]
```diff
@@ -0,0 +1,70 @@
...
+
+import {Image1} from "./img1.webp";
+export {Image1}
+
+
+
+
+
```
Comment (shraddha15-creator) [inline src/assets/Image/ProductListCard/index.js:70]: remove empty lines also

File: src/components/Navbar/Navbar.jsx [hunk 3]
```diff
@@ -0,0 +1,29 @@
+import './Navbar.css';
+import React from "react";
+import {IcRoundFavoriteBorder, IcOutlineShoppingCart, MdiMagnify} from '../../assets/Icon/icon';
+
+import logo from '../../assets/Image/Logo/HIFI.png';
+
+
+
+
```
Comment (shraddha15-creator) [inline src/components/Navbar/Navbar.jsx]: remove empty lines

File: src/components/Navbar/Navbar.jsx [hunk 4]
```diff
@@ -0,0 +1,29 @@
...
+      <div className="nav-search-container">
+        <input class="search-bar" type="search" placeholder="Search" />
+        <MdiMagnify className="nav-search-icon"/>
+      </div>
+      <div class="btn-left">
+        <button class="nav-btn">Login</button>
+       
+        <IcRoundFavoriteBorder className="nav-heart-icon" />
+        <IcOutlineShoppingCart className="nav-cart-icon" />
+        {/* <IcRoundLogout className="nav-logout-icon"/> */}
```
Comment (shraddha15-creator) [inline src/components/Navbar/Navbar.jsx:20]: remove commented code

File: src/components/ProductListCard/ProductListCard.js [hunk 5]
```diff
@@ -0,0 +1,88 @@
+import { MdiStar, MdiCart } from "../../assets/Icon/icon";
+import { useProduct } from "../../Context/productContext";
+// import  img1 from "../../assets/Image/ProductListCard/img1.webp";
```
Comment (shraddha15-creator) [inline src/components/ProductListCard/ProductListCard.js]: remove if its not needed

File: src/components/Sidebar/FinalFilterProduct/FinalFilterProduct.css [hunk 6]
```diff
@@ -0,0 +1,258 @@
...
+  border-radius: 5px;
+  box-shadow: rgba(0, 0, 0, 0.15) 1.95px 1.95px 2.6px;
+}
+
+/* .product-remove {
+    position: absolute;
+    right: 0.5rem;
+    top: 0.5rem;
+    font-size: 2rem;
+  } */
```
Comment (shraddha15-creator) [inline src/components/Sidebar/FinalFilterProduct/FinalFilterProduct.css]: remove if its not needed

File: src/components/Sidebar/FinalFilterProduct/FinalFilterProduct.css [hunk 7]
```diff
@@ -0,0 +1,258 @@
...
+  align-items: center;
+  padding-top: 3rem;
+  padding-bottom: 3rem;
+  font-size: 2rem;
+  font-weight: 700;
+}
+/* .card-container {
+    width: 100%;
+    position: relative;
+  } */
```
Comment (shraddha15-creator) [inline src/components/Sidebar/FinalFilterProduct/FinalFilterProduct.css]: this also

File: src/components/Sidebar/FinalFilterProduct/FinalFilterProduct.js [hunk 8]
```diff
@@ -0,0 +1,84 @@
...
+import "./FinalFilterProduct.css";
+import {products} from "../../../backend/db/products";
+import { ProductListCard } from "../../../components/ProductListCard/ProductListCard";
+import { useProduct } from "../../../Context/productContext";
+import { FilterByBrand } from "../SidebarFilter/FilterByBrand";
+import { FilterByDiscount } from "../SidebarFilter/FilterByDiscount";
+import { FilterByPriceHighLow } from "../SidebarFilter/FilterByPriceHighLow";
+import { FilterByPrice } from "../SidebarFilter/FilterByPrice";
+import { FilterByRating } from "../SidebarFilter/FilterByRating";
+import { FilterByCategory } from "../SidebarFilter/FilterByCategory";
```
Comment (shraddha15-creator) [inline src/components/Sidebar/FinalFilterProduct/FinalFilterProduct.js:11]: You can import all the filter functions in index.js file and import it here

File: src/components/Sidebar/FinalFilterProduct/FinalFilterProduct.js [hunk 9]
```diff
@@ -0,0 +1,84 @@
...
+import axios from 'axios';
+
+export function FinalFilterProduct() {
+  const { filterState } = useProduct();
+  const [productsList, setProductsList] = useState();
+
+  console.log(productsList);
```
Comment (shraddha15-creator) [inline src/components/Sidebar/FinalFilterProduct/FinalFilterProduct.js:20]: remove console

File: src/components/Sidebar/Sidebar.jsx [hunk 10]
```diff
@@ -0,0 +1,48 @@
+import "./Sidebar.css";
+import { useProduct } from "../../Context/productContext";
+import { BrandInput } from "../Sidebar/SidebarInput/BrandInput";
+import { DiscountInput } from "../Sidebar/SidebarInput/DiscountInput";
+import { PriceHighLowInput } from "../Sidebar/SidebarInput/PriceHighLowInput";
+import { PriceInput } from "../Sidebar/SidebarInput/PriceInput";
+import { RatingInput } from "../Sidebar/SidebarInput/RatingInput";
+import { CategoryInput } from "../Sidebar/SidebarInput/CategoryInput";
```
Comment (shraddha15-creator) [inline src/components/Sidebar/Sidebar.jsx:8]: same for this. export it in the index.js file and import it here.

**Review by shraddha15-creator — CHANGES_REQUESTED** [review]

**Review by devsurajrai — APPROVED** [review]
Try writing more abstracted code. For example you can abstract out the API fetch in FinalFilterProduct.js and just deliver the end product. No breaking changes are there. You can proceed ahead.

**Comment by sushmita12345** [comment]
Thanks, changes has been done.

**Pull request merged**